model/src/main/kotlin/CuratedPackage.kt

```diff
@@ -44,7 +44,12 @@ data class CuratedPackage(
     /**
      * A comparison function to sort packages by their identifier.
      */
-    override fun compareTo(other: CuratedPackage) = metadata.id.compareTo(other.metadata.id)
+    override fun compareTo(other: CuratedPackage) =
+        when {
+            metadata.id < other.metadata.id -> -1
+            metadata.id > other.metadata.id -> 1
+            else -> if (this == other) 0 else 1 // Return arbitrary inequality.
+        }
```

Comment:
See this commit, I would again like to bring up the question whether
using a combination of the following two for the task of sorting things in serialization is the right approach:

1. Sorted collections
2. Implementing Comparable

E.g.: If implementing Comparable was avoided (for ordering things in the serializer), would these kind of problems then be not present?

Reply:
Seems we agree that this could be good to do. I've tried to write this down in an issue, see https://github.com/oss-review-toolkit/ort/issues/6235.

Reply:
How do you prefer to move this PR forward?

Reply:
I don't know yet. This is just a draft to play with (I wanted to see what tests it breaks, if any).